src/test/java/uk/gov/companieshouse/api/testdata/service/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,213 @@
+import uk.gov.companieshouse.api.testdata.repository.RoleRepository;
+import uk.gov.companieshouse.api.testdata.repository.UserRepository;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+
+@ExtendWith(MockitoExtension.class)
+class UserServiceImplTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private RoleRepository roleRepository;
+
+    @InjectMocks
+    private UserServiceImpl userServiceImpl;
+
+    @Test
+    void testCreateUserWithoutRoles() throws DataException {
+        UsersSpec usersSpec = new UsersSpec();
+        usersSpec.setPassword("password");
+        UserTestData userTestData = userServiceImpl.create(usersSpec);
+        verify(userRepository).save(argThat(user -> {
+            assertEquals(usersSpec.getPassword(), user.getPassword(), "Password should match the one set in UsersSpec");
+            return true;
+        }));
+        assertNotNull(userTestData.getUserId(), "User ID should not be null");
+        assertNotNull(userTestData.getEmail(), "Email should not be null");
+        assertTrue(userTestData.getForename().contains("Forename"), "Forename should contain Forename");
+        assertTrue(userTestData.getSurname().contains("Surname"), "Surname should contain Surname");
+
+        ArgumentCaptor<Roles> rolesCaptor = ArgumentCaptor.forClass(Roles.class);
+        verify(roleRepository, times(0)).save(rolesCaptor.capture());
+    }
+
+    @Test
+    void testCreateUserWithRoles() throws DataException {
+        UsersSpec usersSpec = new UsersSpec();
+        usersSpec.setPassword("password");
+
+        RolesSpec roleSpec = new RolesSpec();
+        roleSpec.setId("role-id");
+        roleSpec.setPermissions(Arrays.asList("permission1", "permission2"));
+        usersSpec.setRoles(List.of(roleSpec));
+
+        UserTestData userTestData = userServiceImpl.create(usersSpec);
+        ArgumentCaptor<Roles> rolesCaptor = ArgumentCaptor.forClass(Roles.class);
+        verify(roleRepository).save(rolesCaptor.capture());
+
+        Roles capturedRole = rolesCaptor.getValue();
+        assertNotNull(capturedRole, "Captured role should not be null");
+        assertTrue(capturedRole.getId().contains(roleSpec.getId()), "Role ID should match the one set in RolesSpec");
+        assertEquals(roleSpec.getPermissions(), capturedRole.getPermissions(), "Permissions should match the ones set in RolesSpec");
+
+        ArgumentCaptor<Users> usersCaptor = ArgumentCaptor.forClass(Users.class);
+        verify(userRepository).save(usersCaptor.capture());
+
+        Users capturedUser = usersCaptor.getValue();
+        assertNotNull(capturedUser.getRoles(), "User roles should not be null");
+        assertEquals(1, capturedUser.getRoles().size(), "User should have one role assigned");
+        assertEquals(capturedRole.getId(), capturedUser.getRoles().get(0), "The assigned role should match the captured role");
```

Comment:
Can we also assert the rest of values: locale, password, direct login privilege, created...